pkg/csi/service/fcd/controller.go

```diff
@@ -56,6 +58,16 @@ func New() vTypes.Controller {
 }
 
 func (c *controller) Init(config *vcfg.Config) error {
+	glogLevel := os.Getenv(GLOG_LEVEL)
```

Comment:
Hmm. I'm not sure I like having to check for this as well. I see why we are doing it, because the common components are using glog, but I wonder if we could instead trigger the below logic based of the X_CSI_LOG_LEVEL field, and or the X_CSI_DEBUG. Like, `if X_CSI_DEBUG == true; set glog = 5`, stuff like that. Set the glog level automatically.

Reply:
Without setting GLOG to 2, there is absolutely zero logging and no way to debug or even see what is going on.

Reply:
Oh, I agree with you there.  I'm saying rather than support `GLOG_LEVEL` as an env var for the CSI plugin, we do the following:

```go
flag.Set("logtostderr", "true")
flag.Set("stderrthreshold", "INFO")

lvl := log.GetLevel()
if lvl == log.Debug
  flag.Set("v", "5")
if lvl == log.Info
  flag.Set("v", "4")

flag.Parse()
```

Something like that (with whatever is the right mapping from the logrus log levels to the glog levels)

The above psuedocode works (just reading the exsting log level) because GoCSI does [this](https://github.com/rexray/gocsi/blob/9407f5f4ed38592fe11b57c1c9d3621df05b753b/gocsi.go#L39-L56) to set it at the beginning.

Also, I think this code should live at service/service.go:BeforeRun() [here](https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/csi/service/service.go#L94) instead of in fcd/controller.go. It should be a common thing that always happens.

Reply:
Yup, gotcha now... will update